cli/command/formatter/disk_usage.go

```diff
@@ -144,57 +177,91 @@ func (ctx *DiskUsageContext) verboseWrite() error {
 			}
 		}
 
-		err := ctx.contextFormat(tmpl, &imageContext{
+		duc.Images = append(duc.Images, &imageContext{
 			repo:  repo,
 			tag:   tag,
-			trunc: true,
+			trunc: ctx.trunc,
 			i:     *i,
 		})
-		if err != nil {
+	}
+
+	// Now containers
+	for _, c := range ctx.Containers {
+		// Don't display the virtual size
+		c.SizeRootFs = 0
+		duc.Containers = append(duc.Containers, &containerContext{trunc: ctx.trunc, c: *c})
+	}
+
+	// And volumes
+	for _, v := range ctx.Volumes {
+		duc.Volumes = append(duc.Volumes, &volumeContext{v: *v})
+	}
+
+	// And build cache
+	buildCacheSort(ctx.BuildCache)
+	for _, v := range ctx.BuildCache {
+		duc.BuildCache = append(duc.BuildCache, &buildCacheContext{v: v, trunc: ctx.trunc})
+	}
+
+	if ctx.Context.Format == TableFormatKey {
+		return ctx.verboseWriteTable(duc)
+	}
+
+	ctx.preFormat()
+	tmpl, err := ctx.parseFormat()
+	if err != nil {
+		return err
+	}
+	if err := tmpl.Execute(ctx.Output, duc); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (ctx *DiskUsageContext) verboseWriteTable(duc *diskUsageContext) error {
+	tmpl, err := ctx.startSubsection(defaultDiskUsageImageTableFormat)
+	if err != nil {
+		return err
+	}
+	ctx.Output.Write([]byte("Images space usage:\n\n"))
```

Comment:
For here and almost all the file
```go
fmt.Fprint(ctx.Output, "Images space usage:\n\n")
```

Reply:
There are some examples of `fmt.Fprintf` just below 😄 but ok for code moved.

Reply:
@silvin-lubecki `fmt.Fprint != fmt.Fprintf` :) but yeah we can discuss further refactors in https://github.com/docker/cli/pull/1369

Reply:
My point was more to avoid manipulating []byte and use string everywhere